lib/matplotlib/rcsetup.py

```diff
@@ -60,8 +62,28 @@ def __call__(self, s):
                          % (self.key, s, list(six.itervalues(self.valid))))
 
 
+def _listify_validator(scalar_validator):
+    def f(s):
+        if isinstance(s, six.string_types):
+            try:
+                return [scalar_validator(v.strip()) for v in s.split(',')
+                        if v.strip()]
+            except Exception:
+                # Sometimes, a list of colors might be a single string
+                # of single-letter colornames. So give that a shot.
+                return [scalar_validator(v.strip()) for v in s if v.strip()]
+        elif type(s) in (list, tuple):
+            return [scalar_validator(v) for v in s if v]
+        else:
+            msg = "'s' must be of type [ string | list | tuple ]"
+            raise ValueError(msg)
+    f.__doc__ = scalar_validator.__doc__
+    return f
+
```

Comment:
Indeed, they are very similar (so, one of us is going to have to rework their code prior to merging). I should point out a couple of functional differences. First, my code was based off of the `validate_stringlist()` function, where yours was based off of `validate_colorlist()`. These behave slightly differently when there are empty strings in the list. I don't know which one should be considered "correct".

Second, I added a little bit of logic in the parser to allow one to specify a single string that is intended to be converted into a list before validating (i.e., the `cycler('color', 'rgbcmyk')` notation).

As for me implementing it as a private closure versus your public function, I can see merits to both approaches. Years of experience, though, have taught me to -- when in doubt -- make a function/method/attribute private at first, because it is very easy to make it public later. But it is much harder to get rid of it or fix it later if it was public. I also don't see it as a general validator, but more like metaprogramming.